src/averageAges.js

```diff
@@ -15,11 +15,16 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(human => century
+    ? (human.sex === 'm' && Math.ceil(human.died / 100) === century)
+    : human.sex === 'm'
+  );
+
+  const totalAge = men.reduce((sum, person) =>
+    sum + (person.died - person.born), 0);
+  const averageAge = totalAge / men.length;
```

Comment:
You repeat the same logic for calculating the average age each time. So, create a separate function for it